app/src/main/java/co/co/nimblehq/showcases/poly/place/SliceProvider.kt

```diff
@@ -0,0 +1,101 @@
+package co.co.nimblehq.showcases.poly.place
+
+import android.content.ContentResolver
+import android.content.Intent
+import android.net.Uri
+import androidx.slice.Slice
+import androidx.slice.SliceProvider
+import androidx.slice.builders.ListBuilder
+import androidx.slice.builders.SliceAction
+
+class SliceProvider : SliceProvider() {
+    /**
+     * Instantiate any required objects. Return true if the provider was successfully created,
+     * false otherwise.
+     */
+    override fun onCreateSliceProvider(): Boolean {
+        return true
+    }
+
+    /**
+     * Converts URL to content URI (i.e. content://co.co.nimblehq.showcases.poly.place...)
+     */
+    override fun onMapIntentToUri(intent: Intent?): Uri {
+        // Note: implementing this is only required if you plan on catching URL requests.
+        // This is an example solution.
+        var uriBuilder: Uri.Builder = Uri.Builder().scheme(ContentResolver.SCHEME_CONTENT)
+        if (intent == null) return uriBuilder.build()
+        val data = intent.data
+        val dataPath = data?.path
+        if (data != null && dataPath != null) {
+            val path = dataPath.replace("/", "")
+            uriBuilder = uriBuilder.path(path)
+        }
+        val context = context
+        if (context != null) {
+            uriBuilder = uriBuilder.authority(context.packageName)
+        }
```

Comment:
I think we can shortened the code to be like this 🤔 
```suggestion
        uriBuilder = uriBuilder.authority(context?.packageName)
```